bluetooth: fast_pair: Add TX power calibration documentation (#9008)
nrfconnect/sdk-nrf: doc/nrf/ug_bt_fast_pair.rst

See the official `Fast Pair Advertising`_ documentation for detailed information about the requirements related to discoverable and not discoverable advertising.
See :file:`samples/bluetooth/peripheral_fast_pair/src/bt_adv_helper.c` for an example of the implementation.

Advertising TX power
====================

> According to the `Fast Pair TX power`_ documentation, the Fast Pair Seeker must know the Provider's TX power to determine proximity.
> The TX power can be either:
> 
> * provided during model registration,
> * included in the advertising payload.

Reviewer: ```suggestion
The Fast Pair Seeker must know the TX power of the provider to determine proximity. 
The TX power can be included in one of the following ways:

* During model registration
* Advertising payload

See the `Fast Pair TX power`_ documentation for more information.
```

Author: Rephrased this one a bit

Advertising data provider
-------------------------

If your application uses :ref:`bt_le_adv_prov_readme`, you can use the TX power advertising data provider (:kconfig:option:`CONFIG_BT_ADV_PROV_TX_POWER`) to read the advertising TX power from Bluetooth controller and add it to the generated advertising data.
The :kconfig:option:`CONFIG_BT_ADV_PROV_TX_POWER_CORRECTION_VAL` option can be used to define a TX power correction value that is added to the TX power readout included in the advertising data.

> The option can be used to take into account hardware configuration, for example used antenna and device casing.
> See :ref:`peripheral_fast_pair` sample for an example of the TX power advertising provider usage.

Reviewer: ```suggestion
The option can be used to take into account hardware configuration, for example, used antenna and device casing.
See :ref:`peripheral_fast_pair` sample for an example of how to use the TX power advertising provider.
```

> Multiprotocol Service Layer Front-End Module (MPSL FEM)

Reviewer: ```suggestion
Multiprotocol Service Layer front-end module (MPSL FEM)
```

-------------------------------------------------------

> If your application uses MPSL :ref:`nrfxlib:mpsl_fem`, you can use a Frond-End Module power model.

Reviewer: ```suggestion
If your application uses MPSL :ref:`nrfxlib:mpsl_fem`, you can use a front-end module power model.
```

The power model allows you to control the TX power more accurately and compensate, for example, for external conditions.

> See the ``TX power split using models`` section of :ref:`mpsl_fem` documentation for details.

Reviewer: ```suggestion
See the TX power split using models section of the :ref:`mpsl_fem` documentation for more details.
```

See the official `Fast Pair Advertising`_ documentation for detailed information about the requirements related to discoverable and not discoverable advertising.
See :file:`samples/bluetooth/peripheral_fast_pair/src/bt_adv_helper.c` for an example of the implementation.

Advertising TX power
====================

The Fast Pair Seeker must know the TX power of the Provider to determine proximity.
The TX power can be provided in one of the following ways:

> * defined during model registration
> * included in the advertising payload

Reviewer 2: ```suggestion
* Defined during model registration
* Included in the advertising payload
```
